pkg/ttl/ttlworker/task_manager.go

```diff
@@ -597,6 +601,33 @@ func (t *runningScanTask) Context() context.Context {
 	return t.ctx
 }
 
-func (t *runningScanTask) finished() bool {
-	return t.result != nil && t.statistics.TotalRows.Load() == t.statistics.ErrorRows.Load()+t.statistics.SuccessRows.Load()
+func (t *runningScanTask) finished(logger *zap.Logger) bool {
+	if t.result == nil {
+		// Scan task isn't finished
+		return false
+	}
+
+	logger = logger.With(
+		zap.String("jobID", t.JobID),
+		zap.Int64("scanID", t.ScanID),
+		zap.String("table", t.tbl.Name.O),
+	)
+
+	if t.statistics.TotalRows.Load() <= t.statistics.ErrorRows.Load()+t.statistics.SuccessRows.Load() {
+		// All rows are processed.
+		// We use `<=` instead of `==` to make the logic strong to make sure
+		// it also works when statistics are not accurate.
+		logger.Info("mark TTL task finished because all scanned rows are processed")
```

Comment:
How about adding a warning for the case `t.statistics.ErrorRows.Load()+t.statistics.SuccessRows.Load() > t.statistics.TotalRows.Load()`, as we currently don't know when they are not equal.